src/livekit/useECConnectionState.ts

```diff
@@ -89,12 +115,33 @@ export function useECConnectionState(
       (async () => {
         setSwitchingFocus(true);
         await livekitRoom?.disconnect();
-        await livekitRoom?.connect(sfuConfig!.url, sfuConfig!.jwt);
+        await doConnect(
+          livekitRoom!,
+          sfuConfig!,
+          initialAudioEnabled,
+          initialAudioOptions
+        );
       })();
+    } else if (
+      !sfuConfigValid(currentSFUConfig.current) &&
+      sfuConfigValid(sfuConfig)
+    ) {
+      // if we're transitioning from an invalid config to a valid one (ie. connecting)
+      // then do an initial connection, including publishing the microphone track:
+      // livekit (by default) keeps the mic track open when you mute, but if you start muted,
+      // doesn't publish it until you unmute. We want to publish it from the start so we're
+      // always capturing audio: it helps keep bluetooth headsets in the right mode and
+      // mobile browsers to know we're doing a call.
+      doConnect(
```

Comment:
Should there be an `await` here?

Reply:
It wouldn't cause anything to be awaited anyway since it's within a useEffect callback which can't be async. The one above needs to be in an async wrapper to wait for the previous call to finsh before running doConnect().